rayon-core/src/thread_pool/mod.rs

```diff
@@ -50,6 +50,35 @@ impl ThreadPool {
         }
     }
 
+    /// Executes `op` within the threadpool without blocking the
+    /// calling thread. Any attempts to use `join`, `scope`, or
+    /// parallel iterators will then operate within that
+    /// threadpool.
+    ///
+    /// # Warning: thread-local data
+    ///
+    /// Because `op` is executing within the Rayon thread-pool,
+    /// thread-local data from the current thread will not be
+    /// accessible.
+    ///
+    /// # Panics
+    ///
+    /// Polling the [`InstallFuture`] will propagate panics from
+    /// `op`.
+    ///
+    /// [`InstallFuture`]: struct.InstallFuture.html
+    #[cfg(feature = "unstable")]
```

Comment:
Is it good to place it here, or would `feature != unstable` add these doc comments to `current_num_threads`?